File: shanny-blog-java/blog-server/src/main/java/com/shanny/controller/AboutController.java
package com.shanny.controller;

import com.shanny.dto.AboutDTO;
import com.shanny.dto.ArticleDTO;
import com.shanny.result.Result;
import com.shanny.service.AboutService;
import com.shanny.vo.AboutVO;
import com.shanny.vo.ArticleVO;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController()
@RequestMapping("/about")
@Tag(name = "菜单相关接口")
@Slf4j
public class AboutController {
    private final AboutService aboutService;

    @Autowired
    public AboutController(AboutService aboutService) {
        this.aboutService = aboutService;
    }

    @GetMapping("/all")
    @Operation(summary = "作者信息获取")
    public Result<List<AboutVO>> GetAboutMe() {
        try {
            return aboutService.getAboutMe();
        } catch (Exception e) {
            return Result.error(e.getMessage());
        }
    }

    @GetMapping("/show")
    @Operation(summary = "作者信息获取")
    public Result<AboutVO> GetAboutMeByShow() {
        try {
            return aboutService.getAboutMeByShow();
        } catch (Exception e) {
            return Result.error(e.getMessage());
        }
    }

    @PostMapping("/add")
    @Operation(summary = "作者信息添加")
    public Result<AboutVO> AddAboutMe(@RequestBody AboutDTO aboutDTO) {
        try {
            return aboutService.addAbout(aboutDTO);
        } catch (Exception e) {
            return Result.error(e.getMessage());
        }
    }

    @PostMapping("/update")
    @Operation(summary = "作者信息修改")
    public Result<AboutVO> UpdateAboutMe(@RequestBody AboutDTO aboutDTO) {
        try {
            return aboutService.updateAbout(aboutDTO);
        } catch (Exception e) {
            return Result.error(e.getMessage());
        }
    }

    @PostMapping("/delete")
    @Operation(summary = "作者信息删除")
    public Result<String> deleteAboutMe(Long id) {
        try {
            return aboutService.deleteAboutById(id);
        } catch (Exception e) {
            return Result.error(e.getMessage());
        }
    }
}

File: shanny-blog-net/blog-server/Controllers/AboutController.cs
using blog_common.Result;
using blog_pojo.Dtos;
using blog_pojo.Vos;
using Microsoft.AspNetCore.Mvc;

namespace blog_server.Controllers
{
    [ApiController]
    [Route("/about")]
    [Tags("菜单相关接口")]
    public class AboutController : ControllerBase
    {
        private readonly AboutService _aboutService;
        private readonly ILogger<AboutController> _log;

        public AboutController(AboutService aboutService, ILogger<AboutController> log)
        {
            _aboutService = aboutService;
            _log = log;
        }

        /// <summary>
        /// 作者信息获取
        /// </summary>
        /// <returns></returns>
        [HttpGet("all")]
        [ProducesResponseType(typeof(Result<List<AboutVO>>), 200)]
        public Result<List<AboutVO>> GetAboutMe()
        {
            try
            {
                return _aboutService.GetAboutMe();
            }
            catch (Exception e)
            {
                return Result<List<AboutVO>>.Error(e.Message);
            }
        }

        /// <summary>
        /// 作者信息获取
        /// </summary>
        [HttpGet("show")]
        [ProducesResponseType(typeof(Result<AboutVO>), 200)]
        public Result<AboutVO> GetAboutMeByShow()
        {
            try
            {
                return _aboutService.GetAboutMeByShow();
            }
            catch (Exception e)
            {
                return Result<AboutVO>.Error(e.Message);
            }
        }

        /// <summary>
        /// 作者信息添加
        /// </summary>
        [HttpPost("add")]
        [ProducesResponseType(typeof(Result<AboutVO>), 200)]
        public Result<AboutVO> AddAboutMe([FromBody] AboutDTO aboutDTO)
        {
            try
            {
                return _aboutService.AddAbout(aboutDTO);
            }
            catch (Exception e)
            {
                return Result<AboutVO>.Error(e.Message);
            }
        }

        /// <summary>
        /// 作者信息修改
        /// </summary>
        [HttpPost("update")]
        [ProducesResponseType(typeof(Result<AboutVO>), 200)]
        public Result<AboutVO> UpdateAboutMe([FromBody] AboutDTO aboutDTO)
        {
            try
            {
                return _aboutService.UpdateAbout(aboutDTO);
            }
            catch (Exception e)
            {
                return Result<AboutVO>.Error(e.Message);
            }
        }

        /// <summary>
        /// 作者信息删除
        /// </summary>
        [HttpPost("delete")]
        [ProducesResponseType(typeof(Result<string>), 200)]
        public Result<string> DeleteAboutMe(long id)
        {
            try
            {
                return _aboutService.DeleteAboutById(id);
            }
            catch (Exception e)
            {
                return Result<string>.Error(e.Message);
            }
        }
    }
}
